### DISCUSSION TITLE
Anyone has good way to exclude api key from repo?

### DISCUSSION THREAD

**Discussion by soemarko** [react-native-maps/react-native-maps#4579] [Q&A] [answered]
At least for Android (I'm using Apple map on iOS). I think it's unreasonable to add `AndroidManifest.xml` into gitignore. What are my options here?

Ideas for iOS side could be added here too, to help others that come here in the future.

**Comment by jan-kozinski** [comment]
You can use android/local.properties or ios/Config.xcconfig files to store your API keys

**Reply by dev-externlabs** [reply]
How would you access them in `AndroidManifest.xml`?

**Reply by jan-kozinski** [reply, marked as answer]
Hi, sorry for not responding for a while. You can access them by adding 
```
<meta-data
          android:name="com.google.android.geo.API_KEY"
          android:value="${MAPS_API_KEY}"/>
```
to the contents of application tag. See https://github.com/react-native-maps/react-native-maps/blob/master/example/android/app/src/main/AndroidManifest.xml for reference

**Reply by dev-externlabs** [reply]
thisi is in my local.properties
```
## This file must *NOT* be checked into Version Control Systems,
# as it contains information specific to your local configuration.
#
# Location of the SDK. This is only used by Gradle.
# For customization when using a Version Control System, please read the
# header note.
#Mon Oct 30 11:04:36 IST 2023
sdk.dir=/Users/devendragaud/Library/Android/sdk
MAPS_API_KEY=SomESEcretValue
```
and this is in my manifest.xml
```
 <meta-data
            android:name="com.google.android.geo.API_KEY"
            android:value="${MAPS_API_KEY}"/>
```
and it give error

`/Users/devendragaud/projects/Leaks/leaks_mobile/android/app/src/main/AndroidManifest.xml:28:13-44 Error:
	Attribute meta-data#com.google.android.geo.API_KEY@value at AndroidManifest.xml:28:13-44 requires a placeholder substitution but no value for <MAPS_API_KEY> is provided.`

**Reply by dev-externlabs** [reply]
I also had to add this

    Properties properties = new Properties()
    properties.load(project.rootProject.file('local.properties').newDataInputStream())
    manifestPlaceholders = [MAPS_API_KEY: "${properties.getProperty('MAPS_API_KEY')}"]
   
in my `build.gradle(android\app`'s `default.config`
